ts/components/screens/OperationResultScreenContent.tsx

```diff
@@ -58,24 +111,45 @@ const OperationResultScreenContent = ({
       {subtitle && (
         <>
           <VSpacer size={8} />
-          <LabelSmall style={styles.text} color="grey-650" weight="Regular">
-            {subtitle}
-          </LabelSmall>
+          {typeof subtitle === "string" ? (
+            <Body style={styles.text}>{subtitle}</Body>
+          ) : (
+            <ComposedBodyFromArray subtitle={subtitle} textAlign="center" />
+          )}
         </>
       )}
-      {action && (
+      {action &&
+        (!action.buttonType || action.buttonType === "ButtonSolid") && (
+          <View style={IOStyles.alignCenter}>
+            <VSpacer size={24} />
+            <View>
+              <ButtonSolid {...(action as ButtonSolidProps)} />
+            </View>
+          </View>
+        )}
+      {action && action.buttonType === "ButtonLink" && (
         <View style={IOStyles.alignCenter}>
           <VSpacer size={24} />
           <View>
-            <ButtonSolid {...action} />
+            <ButtonLink {...action} />
           </View>
         </View>
       )}
-      {secondaryAction && (
+      {secondaryAction &&
+        (!secondaryAction.buttonType ||
+          secondaryAction.buttonType === "ButtonLink") && (
+          <View style={IOStyles.alignCenter}>
+            <VSpacer size={24} />
+            <View>
+              <ButtonLink {...secondaryAction} />
+            </View>
+          </View>
+        )}
+      {secondaryAction && secondaryAction.buttonType === "ButtonSolid" && (
         <View style={IOStyles.alignCenter}>
           <VSpacer size={24} />
           <View>
-            <ButtonLink {...secondaryAction} />
+            <ButtonSolid {...(secondaryAction as ButtonSolidProps)} />
           </View>
         </View>
       )}
```

Comment:
What about this?

```ts
      {action && (
        <View style={IOStyles.alignCenter}>
          <VSpacer size={24} />
          <View>
            {action.buttonType === "ButtonLink" ? (
              <ButtonLink {...action} />
            ) : (
              <ButtonSolid {...(action as ButtonSolidProps)} />
            )}
          </View>
        </View>
      )}
      {secondaryAction && (
        <View style={IOStyles.alignCenter}>
          <VSpacer size={24} />
          <View>
            {secondaryAction.buttonType === "ButtonSolid" ? (
              <ButtonSolid {...(secondaryAction as ButtonSolidProps)} />
            ) : (
              <ButtonLink {...secondaryAction} />
            )}
          </View>
        </View>
      )}
```

What do you think?

Reply:
as a consequence of changes requested by the design team, I deleted this part of the code.

Reply:
Do you mean that action is always a solid button and secondAction is always a link button? Am I right?